tests/app/shared/issue/duplicatedIssues/duplicated-issues.component.spec.ts

```diff
@@ -0,0 +1,164 @@
+    let nativeElement: HTMLElement;
+    let fixture: ComponentFixture<DuplicatedIssuesComponent>;
+    let dummyDuplicatedIssues: Observable<Issue[]>;
+    let thisIssue: Issue;
+
+    const dummyTeam = TEAM_4;
+    const firstDummyIssue =  Issue.createPhaseModerationIssue(ISSUE_PENDING_MODERATION_HIGH_SEVERITY_FEATURE_FLAW, dummyTeam);
+    const secondDummyIssue = Issue.createPhaseModerationIssue(ISSUE_PENDING_MODERATION_LOW_SEVERITY_DOCUMENTATION_BUG, dummyTeam);
+
+    const permissionService: any = jasmine.createSpyObj('PermissionService', ['isTeamResponseEditable', 'isTutorResponseEditable']);
+    const issueService: any = jasmine.createSpyObj('IssueService', ['getDuplicateIssuesFor', 'updateIssueWithComment', 'updateIssue', 'updateLocalStore']);
+    const phaseService: any = jasmine.createSpyObj('PhaseService', [], { currentPhase: Phase.phaseModeration });
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                DuplicatedIssuesComponent
+            ],
+            providers: [
+                IssueService, ErrorHandlingService, PhaseService, PermissionService, {
+                    provide: HAMMER_LOADER,
+                    useValue: () => new Promise(() => {})
+                  }
+            ],
+            imports: [
+                MaterialModule, RouterTestingModule, BrowserDynamicTestingModule, ApolloTestingModule, HttpClientTestingModule
+            ]
+        })
+        .overrideProvider(PermissionService, { useValue: permissionService })
+        .overrideProvider(IssueService, { useValue: issueService})
+        .overrideProvider(PhaseService, { useValue: phaseService})
+        .compileComponents();
+    }));
+
+    beforeEach(() => {
+        dummyDuplicatedIssues = of([firstDummyIssue, secondDummyIssue]);
+        issueService.getDuplicateIssuesFor.and.callFake(() => dummyDuplicatedIssues);
+
+        thisIssue = Issue.createPhaseModerationIssue(ISSUE_PENDING_MODERATION, dummyTeam);
+        firstDummyIssue.duplicateOf = thisIssue.id;
+        secondDummyIssue.duplicateOf = thisIssue.id;
+
+        fixture = TestBed.createComponent(DuplicatedIssuesComponent);
+        component = fixture.componentInstance;
+        component.issue = thisIssue;
+        fixture.detectChanges();
+
+        debugElement = fixture.debugElement;
+        nativeElement = fixture.nativeElement;
+    });
+
+    it('should create a chip for each duplicated issue', () => {
+        // Check that the number of chips is the same as the number of duplicate issues
+        const matChipListLength: number = debugElement.queryAll(By.css('.mat-chip')).length;
+        const duplicateIssueslength: number = getDuplicateIssuesLength();
+        expect(matChipListLength).toEqual(duplicateIssueslength);
+
+        // Check that first chip contains information on first issue in the duplicate issues array
+        const matChipAnchor: HTMLElement = nativeElement.querySelector('a');
```

Comment:
Just one small nitpick I missed in my earlier review. 

Maybe we can consider using `queryAll` to check for the number of links with respect to the number of `duplicateIssues` currently present!

Reply:
I thought perhaps that would be redundant, since I am already checking the length of the `matChipListLength`! Which is why I am only checking that the content of the first `mat-chip-anchor` contains the ID/link of the first dummy issue, to make sure the links are displayed together with the `mat-chip`. Do you think I should still include the number of links?

Reply:
Then I think this will be fine ba